### ISSUE TITLE
error @:build

### ISSUE THREAD

**Issue by WeyskoteFoundation** [TiVo/intellij-haxe#114]
![haxe](https://cloud.githubusercontent.com/assets/4976223/6014335/d0c8aeba-ab48-11e4-8f62-75de9ae9b14d.png)

https://github.com/andyli/jQueryExternForHaxe/blob/master/jQuery/JQuery.hx

**Comment by WeyskoteFoundation** [comment]
![haxe](https://cloud.githubusercontent.com/assets/4976223/6014346/0fdeafd2-ab49-11e4-92c7-e63e6f0e2993.png)

https://github.com/andyli/jQueryExternForHaxe/blob/master/jQuery/JqXHR.hx

**Comment by EBatTiVo** [comment]
@WeyskoteFoundation I appreciate the bugs.  Can you please post code instead of images (even cut'n'paste, is better than images)?  Minimal example code showing exactly the bug, so that we don't have to translate into a test of our own is even better.   The less typing/inventing we have to do, the quicker your problems can be addressed.   Thanks!!

**Comment by EBatTiVo** [comment]
@:build doesn't appear to the the problem here.  I see that the code within the macro is unresolved, not the macro itself giving an error.  Did I miss something?

**Comment by WeyskoteFoundation** [comment]
![haxe](https://cloud.githubusercontent.com/assets/4976223/6014605/960ba7f4-ab4e-11e4-8661-bc989aea1b61.png)

//CODE file main.hx

package weyskote;
@:build(weyskote.TypeBuildingMacro.build("myFunc"))
class Main {
    static public function main() {
        trace(Main.myFunc); // my default
    }
}
//CODE file TypeBuildingMacro.hx
package weyskote;
import haxe.macro.Context;
import haxe.macro.Expr;

class TypeBuildingMacro {
    macro static public function
    build(fieldName:String):Array<Field> {
        var fields = Context.getBuildFields();
        var newField = {
        name: fieldName,
        doc: null,
        meta: [],
        access: [AStatic, APublic],
        kind: FVar(macro : String,
        macro "my default"),
        pos: Context.currentPos()
        };
        fields.push(newField);
        return fields;
    }
}

**Comment by WeyskoteFoundation** [comment]
it does not see @:build(weyskote.TypeBuildingMacro.build("myFunc"))....
it is as if the method does not exist (build("myFunc"))....
addition to the keyword "macro" that  is not identified

**Comment by WeyskoteFoundation** [comment]
http://haxe.org/manual/macro-type-building.html

**Comment by EBatTiVo** [comment]
This appears to be working in the build 0.8.1.1.TiVo.ClassHierarchy.11.  What build are you using? 

(There are other errors marked, but that's probably because this example is part of a larger project:
```haxe
        trace(Main.myFunc); // << myFunc is unresolved b/c we don't run macros in the editor
```
```AStatic```,```APublic```, and ```FVar``` are unresolved,

```haxe
    kind: FVar(macro : String,   // <<-- 'macro' used as a variable name?
```

**Comment by WeyskoteFoundation** [comment]
because it does not identify the function ?
weyskote.TypeBuildingMacro.build("myFunc")

**Comment by EBatTiVo** [comment]
> because it does not identify the function ?

Correct, because the field 'myFunc' doesn't exist in any parseable form prior to executing the macro; the name exists solely as a string argument to the build function.  The IDE will probably never execute macros.  But, hey, if you want to try, I'd recommend taking a look at issue #7, and once that's working, figure out how to make some sort of "ghost" entries in the PSI/AST that can be looked up via a HaxeReference.

**Comment by WeyskoteFoundation** [comment]
but need not run this function only shows that there was already good;

2015-02-04 20:12 GMT-02:00 Eric B <notifications@github.com>:

**Comment by EBatTiVo** [comment]
> but need not run this function only shows that there was already good;

I don't understand what you are saying.  

I think I may have answered a question you weren't asking: Why is "myFunc" still unresolved?

If you were asking whether ```weyskote.TypeBuildingMacro.build(...)``` is resolving, the answer is "Yes, the TiVo.CH.11 build is resolving ```build``` correctly and does not show an error."

![image](https://cloud.githubusercontent.com/assets/7231977/6055077/b3cf6190-acac-11e4-85c6-0ddf30edc0f1.png)
